service_provider.go

```diff
@@ -91,6 +93,18 @@ type ServiceProvider struct {
 	// IDPMetadata is the metadata from the identity provider.
 	IDPMetadata *EntityDescriptor
 
+	// IDPCertificateFingerprint is fingerprint of the idp public certificate. If this field is specified,
+	// IDPCertificateFingerprintAlgorithm must also be specified, and IDPCertificate must not be specified.
+	IDPCertificateFingerprint *string
+	// IDPCertificateFingerprintAlgorithm is fingerprint algorithm used to obtain fingerprint of the idp public
+	// certificate.
+	// If this field is specified, IDPCertificateFingerprint must also be specified, and IDPCertificate must not be specified.
+	IDPCertificateFingerprintAlgorithm *string
+
+	// IDPCertificate to use as idp public certificate. If this field is specified, IDPCertificateFingerprint and
+	// IDPCertificateFingerprintAlgorithm must not be specified.
+	IDPCertificate *string
```

Comment:
```suggestion
	IDPCertificate *x509.Certificate
```

Firstly, thank you for this PR. This functionality is something I also want from this package!

Just a thought, maybe the type here should just be a `x509.Certificate` and the consumer should be responsible for parsing their certificate themselves.

The reason I suggest this is that it's not certain that everyone is going to be using base64-encoded ASN.1 DER strings for their certificates (which it appears from `parseCert` yours does).

In my case, my (legacy) application simply has them encoded as PEM strings like so: `-----BEGIN CERTIFICATE-----\r\nABCEDFG\r\n-----END CERTIFICATE-----`, so to use your PR I would likely need to use `pem.Decode`, then base-64 encode the string, just to provide it to this package, which then base64-decodes it again.

I think this change would make it more reusable.